Alliance.Client/Extensions/ExNativeUI/AgentStatus/ViewModels/AgentInteractionInterfaceVMCustom.cs

```diff
@@ -0,0 +1,616 @@
+			TextObject actionMessage = item.GetActionMessage(weaponToReplaceOnQuickAction, fillUp);
+			TextObject descriptionMessage = item.GetDescriptionMessage(fillUp);
+			if (TextObject.IsNullOrEmpty(actionMessage) || TextObject.IsNullOrEmpty(descriptionMessage))
+			{
+				return;
+			}
+
+			FocusType = 0;
+			if (isInteractable)
+			{
+				MBTextManager.SetTextVariable("USE_KEY", HyperlinkTexts.GetKeyHyperlinkText(HotKeyManager.GetHotKeyId("CombatHotKeyCategory", 13)));
+				if (canQuickPickup)
+				{
+					// Check if Item have a race restriction and display a message if needed
+					Agent main = Agent.Main;
+					ItemObject objecttampon = item.WeaponCopy.Item;
+					ExtendedItem itemEx = objecttampon.GetExtendedItem();
+					IsRaceConditionRespected(main, itemEx);
+					if (!_IsRaceConditionRespected && itemEx != null && itemEx.Race_condition != null && itemEx.Race_condition != "" && Config.Instance.EnableRaceRestrictionOnStuff)
+					{
+						PrimaryInteractionMessage = ($"This item can only be used by {itemEx.Race_condition}").ToString();
+						
+					}
+					else {
+						if (main != null && main.CanInteractableWeaponBePickedUp(item))
+						{
+
+
+							PrimaryInteractionMessage = actionMessage.ToString();
+							MBTextManager.SetTextVariable("KEY", GameTexts.FindText("str_ui_agent_interaction_use"));
+							MBTextManager.SetTextVariable("ACTION", GameTexts.FindText("str_select_item_to_replace"));
+							SecondaryInteractionMessage = GameTexts.FindText("str_hold_key_action").ToString() + GetWeaponSpecificText(item);
+						}
+						else
+						{
+							PrimaryInteractionMessage = actionMessage.ToString();
+							MBTextManager.SetTextVariable("STR1", descriptionMessage);
+							MBTextManager.SetTextVariable("STR2", GetWeaponSpecificText(item));
+							SecondaryInteractionMessage = GameTexts.FindText("str_STR1_space_STR2").ToString();
+						}
+					}
+				}
+				else
+				{
+					MBTextManager.SetTextVariable("KEY", GameTexts.FindText("str_ui_agent_interaction_use"));
+					MBTextManager.SetTextVariable("ACTION", GameTexts.FindText("str_select_item_to_replace"));
+					PrimaryInteractionMessage = GameTexts.FindText("str_hold_key_action").ToString();
+					MBTextManager.SetTextVariable("STR1", descriptionMessage);
+					MBTextManager.SetTextVariable("STR2", GetWeaponSpecificText(item));
+					SecondaryInteractionMessage = GameTexts.FindText("str_STR1_space_STR2").ToString();
+				}
+			}
+			else
+			{
+				PrimaryInteractionMessage = item.GetInfoTextForBeingNotInteractable(Agent.Main).ToString();
+				MBTextManager.SetTextVariable("STR1", descriptionMessage);
+				MBTextManager.SetTextVariable("STR2", GetWeaponSpecificText(item));
+				SecondaryInteractionMessage = GameTexts.FindText("str_STR1_space_STR2").ToString();
```

Comment:
I can see a pattern here. these codes should maybe called a method that will set what you want instead of doing nearly the same work 4 times.

(If you do an abstract parent class (See my comment), this method can be in the parent class then)

Reply:
Same unchanged decompile TaleWorlds code